Link error on Juwels

@akuesters  observed that building the Arbor validation test executable on Juwels gives a link error.

This can be recreated in the default environment:
```
$ ml CMake
$ mkdir b
$ cd b
$ cmake ..
 The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Check for working C compiler: /gpfs/software/juwels/stages/2019a/software/GCCcore/8.3.0/bin/cc
-- Check for working C compiler: /gpfs/software/juwels/stages/2019a/software/GCCcore/8.3.0/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /gpfs/software/juwels/stages/2019a/software/GCCcore/8.3.0/bin/c++
-- Check for working CXX compiler: /gpfs/software/juwels/stages/2019a/software/GCCcore/8.3.0/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
[...]
$ make -j validate
Scanning dependencies of target gtest
Scanning dependencies of target generate_version_hpp
Scanning dependencies of target default_catalogue_cpp_target
Scanning dependencies of target libmodcc
[  0%] Generating _always_rebuild
[  2%] Generating default_catalogue.cpp
[  4%] Generating version.hpp-test
[  6%] Building CXX object test/CMakeFiles/gtest.dir/gtest-all.cpp.o
[  6%] Built target default_catalogue_cpp_target
[  6%] Building CXX object modcc/CMakeFiles/libmodcc.dir/blocks.cpp.o
[...]
[100%] Linking CXX executable ../../bin/validate
CMakeFiles/validate.dir/validate_kinetic.cpp.o(.rodata.cst8+0x18): error: relocation refers to local symbol "_ZNSt17_Function_handlerIFbN3arb16cell_member_typeEEZNS0_L9one_probeES1_EUlS1_E_E9_M_invokeERKSt9_Any_dataOS1_" [21], which is defined in a discarded section
  section group signature: "_ZN3arb23convergence_test_runnerIfE3runERNS_10simulationEffffRKSt6vectorIfSaIfEE"
  prevailing definition is from CMakeFiles/validate.dir/validate_soma.cpp.o
CMakeFiles/validate.dir/validate_synapses.cpp.o(.rodata.cst8+0x40): error: relocation refers to local symbol "_ZNSt17_Function_handlerIFbN3arb16cell_member_typeEEZNS0_L9one_probeES1_EUlS1_E_E9_M_invokeERKSt9_Any_dataOS1_" [30], which is defined in a discarded section
  section group signature: "_ZN3arb23convergence_test_runnerIiE3runERNS_10simulationEifffRKSt6vectorIfSaIfEE"
  prevailing definition is from CMakeFiles/validate.dir/validate_ball_and_stick.cpp.o
collect2: error: ld returned 1 exit status
make[3]: *** [bin/validate] Error 1
make[2]: *** [test/validation/CMakeFiles/validate.dir/all] Error 2
make[1]: *** [test/validation/CMakeFiles/validate.dir/rule] Error 2
make: *** [validate] Error 2
```

Googling the `relocation refers to local symbol` error seems to indicate that it might be a bug in a recent version of the linker. I haven't spent enough time on this to successfully reproduce the error in a smaller example.

Changing `arb::one_probe` from `static inline` to `inline` is sufficient to make the problem go away (and I'll submit a PR shortly.)

--- a/arbor/include/arbor/sampling.hpp
+++ b/arbor/include/arbor/sampling.hpp
@@ -12,7 +12,7 @@ using cell_member_predicate = std::function<bool (cell_member_type)>;
 
 static cell_member_predicate all_probes = [](cell_member_type pid) { return true; };
 
-static inline cell_member_predicate one_probe(cell_member_type pid) {
+inline cell_member_predicate one_probe(cell_member_type pid) {
     return [pid](cell_member_type x) { return pid==x; };
 }
 

